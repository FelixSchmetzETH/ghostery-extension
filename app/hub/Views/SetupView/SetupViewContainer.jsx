/**
 * Setup View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import SetupView from './SetupView';

// Component Views
import SetupBlockingView from '../SetupViews/SetupBlockingView';
import SetupAntiSuiteView from '../SetupViews/SetupAntiSuiteView';
import SetupHumanWebView from '../SetupViews/SetupHumanWebView';
import SetupDoneView from '../SetupViews/SetupDoneView';

/**
 * @class Implement the Setup Container View for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class SetupViewContainer extends React.Component {
	/**
	 * Lifecycle Event
	 */
	componentWillMount() {
		const {
			blockingPolicy,
			enable_anti_tracking,
			enable_ad_block,
			enable_smart_blocking,
			enable_ghostery_rewards,
			enable_human_web,
		} = this.props;
		const title = 'Ghostery Hub - Setup';

		window.document.title = title;
		this.props.actions.initSetupProps({
			blockingPolicy,
			enable_anti_tracking,
			enable_ad_block,
			enable_smart_blocking,
			enable_ghostery_rewards,
			enable_human_web,
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Setup View of the Hub app
	 */
	render() {
		const activeIndex = +this.props.location.pathname.split('/').pop();
		const steps = [
			{
				index: 1,
				path: '/setup/1',
				bodyComponent: SetupBlockingView,
				headerProps: {
					title: 'Ghostery can block trackers to make your browsing cleaner, faster, and safer. What would you like to block?',
					titleImage: '/app/images/hub/setup/ghosty-block-all.svg',
				},
			},
			{
				index: 2,
				path: '/setup/2',
				bodyComponent: SetupAntiSuiteView,
				headerProps: {
					title: 'Would you like to browse smarter with our recommended features?',
					titleImage: '/app/images/hub/setup/ghosty-shield-stop-lightbulb.svg',
				},
			},
			{
				index: 3,
				path: '/setup/3',
				bodyComponent: SetupHumanWebView,
				headerProps: {
					title: 'Would you like to help improve Ghostery?',
					titleImage: '/app/images/hub/setup/ghosty-human-web.svg',
				},
			},
			{
				index: 4,
				path: '/setup/4',
				bodyComponent: SetupDoneView,
				headerProps: {
					title: 'You are ready!',
					titleImage: '/app/images/hub/setup/ghosty-check-wrench.svg',
				},
			},
		];

		return <SetupView activeIndex={activeIndex} steps={steps} />;
	}
}

// Default props used throughout the Setup flow
SetupViewContainer.defaultProps = {
	blockingPolicy: 'recommended',
	enable_anti_tracking: true,
	enable_ad_block: true,
	enable_smart_blocking: true,
	enable_ghostery_rewards: true,
	enable_human_web: true,

	actions: { // ToDo: remove this
		initSetupProps: () => {},
	},
	location: { // ToDo: remove this
		pathname: '/setup/1',
	},
};

export default SetupViewContainer;
