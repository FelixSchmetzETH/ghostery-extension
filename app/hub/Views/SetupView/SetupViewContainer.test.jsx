/**
 * Setup View Container Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file
 */

import React from 'react';
import renderer from 'react-test-renderer';
// import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

// Mock Necessary Imports
jest.mock('../SetupViews/SetupBlockingView', () => props => <div>Mock Setup Blocking View</div>);
jest.mock('../SetupViews/SetupAntiSuiteView', () => props => <div>Mock Setup Anti-Suite View</div>);
jest.mock('../SetupViews/SetupHumanWebView', () => props => <div>Mock Setup Human Web View</div>);
jest.mock('../SetupViews/SetupDoneView', () => props => <div>Mock Setup Done View</div>);
jest.mock('../../SteppedNavigation', () => props => <div>Mock Stepped Navigation</div>);

// Import Components
import SetupViewContainer from './SetupViewContainer';
// import SetupView from './SetupView';
// import SetupHeader from '../SetupViews/SetupHeader';

// Import Mocked Setup Views
// import SetupBlockingView from '../SetupViews/SetupBlockingView';
// import SetupAntiSuiteView from '../SetupViews/SetupAntiSuiteView';
// import SetupHumanWebView from '../SetupViews/SetupHumanWebView';
// import SetupDoneView from '../SetupViews/SetupDoneView';
// import SteppedNavigation from '../../SteppedNavigation';

describe('app/hub/Views/SetupView container', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup view container is rendered correctly on the Blocking step', () => {
			const paths = ['/setup/1', '/setup/2', '/setup/3', '/setup/4'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={0} >
					<SetupViewContainer />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('setup view container is rendered correctly on the Anti-Suite step', () => {
			const paths = ['/setup/1', '/setup/2', '/setup/3', '/setup/4'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={1} >
					<SetupViewContainer />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('setup view container is rendered correctly on the Human Web step', () => {
			const paths = ['/setup/1', '/setup/2', '/setup/3', '/setup/4'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={2} >
					<SetupViewContainer />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('setup view container is rendered correctly on the Done step', () => {
			const paths = ['/setup/1', '/setup/2', '/setup/3', '/setup/4'];
			const component = renderer.create(
				<MemoryRouter initialEntries={paths} initialIndex={3} >
					<SetupViewContainer />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Mount snapshot tests rendered with Enzyme', () => {
		test.skip('the happy path of the component', () => {
			const paths = ['/setup/1', '/setup/2', '/setup/3', '/setup/4'];
			const component = mount(
				<MemoryRouter initialEntries={paths} initialIndex={0} >
					<SetupViewContainer />
				</MemoryRouter>
			);
			expect(component.find(SetupViewContainer).length).toBe(1);
			expect(component.find(SetupView).length).toBe(1);
			expect(component.find(SetupHeader).length).toBe(1);
			expect(component.find(SteppedNavigation).length).toBe(1);
			expect(component.find(SetupBlockingView).length).toBe(1);
			expect(component.find(SetupAntiSuiteView).length).toBe(0);
			expect(component.find(SetupHumanWebView).length).toBe(0);
			expect(component.find(SetupDoneView).length).toBe(0);
			//
			// component.setProps({ initialIndex: 1 });
			// expect(component.find(SetupView).length).toBe(1);
			// expect(component.find(SetupHeader).length).toBe(1);
			// expect(component.find(TestComponent).length).toBe(1);
			// expect(component.find(SteppedNavigation).length).toBe(1);
			// expect(component.find({ src: 'title-image-1'}).length).toBe(0); // Broken because router doesn't automatically update activeIndex
			// expect(component.find({ src: 'title-image-2'}).length).toBe(1); // Broken because router doesn't automatically update activeIndex
		});

		test.skip('the non-happy path of the component', () => {
		});
	});
});
