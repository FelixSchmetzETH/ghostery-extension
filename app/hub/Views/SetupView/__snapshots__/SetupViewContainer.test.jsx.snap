// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupView container Snapshot tests with react-test-renderer setup view container is rendered correctly on the Anti-Suite step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div
        className="row align-center"
      >
        <div
          className="columns"
        >
          <div
            className="row align-middle align-center"
          >
            <div
              className="columns shrink"
            >
              <img
                src="/app/images/hub/setup/ghosty-shield-stop-lightbulb.svg"
              />
            </div>
            <div
              className="columns shrink"
            >
              <h3
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Would you like to browse smarter with our recommended features?",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        Mock Setup Anti-Suite View
      </div>
    </div>
  </div>
  <div>
    Mock Stepped Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView container Snapshot tests with react-test-renderer setup view container is rendered correctly on the Blocking step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div
        className="row align-center"
      >
        <div
          className="columns"
        >
          <div
            className="row align-middle align-center"
          >
            <div
              className="columns shrink"
            >
              <img
                src="/app/images/hub/setup/ghosty-block-all.svg"
              />
            </div>
            <div
              className="columns shrink"
            >
              <h3
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Ghostery can block trackers to make your browsing cleaner, faster, and safer. What would you like to block?",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        Mock Setup Blocking View
      </div>
    </div>
  </div>
  <div>
    Mock Stepped Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView container Snapshot tests with react-test-renderer setup view container is rendered correctly on the Done step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div
        className="row align-center"
      >
        <div
          className="columns"
        >
          <div
            className="row align-middle align-center"
          >
            <div
              className="columns shrink"
            >
              <img
                src="/app/images/hub/setup/ghosty-check-wrench.svg"
              />
            </div>
            <div
              className="columns shrink"
            >
              <h3
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "You are ready!",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        Mock Setup Done View
      </div>
    </div>
  </div>
  <div>
    Mock Stepped Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView container Snapshot tests with react-test-renderer setup view container is rendered correctly on the Human Web step 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div
        className="row align-center"
      >
        <div
          className="columns"
        >
          <div
            className="row align-middle align-center"
          >
            <div
              className="columns shrink"
            >
              <img
                src="/app/images/hub/setup/ghosty-human-web.svg"
              />
            </div>
            <div
              className="columns shrink"
            >
              <h3
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Would you like to help improve Ghostery?",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        Mock Setup Human Web View
      </div>
    </div>
  </div>
  <div>
    Mock Stepped Navigation
  </div>
</div>
`;
